frontend/src/metabase/query_builder/components/GuiQueryEditor.jsx

```diff
@@ -162,10 +196,10 @@ export default class GuiQueryEditor extends Component {
                     <AggregationWidget
                         key={"agg"+index}
                         aggregation={aggregation}
-                        tableMetadata={tableMetadata}
-                        customFields={Query.getExpressions(this.props.datasetQuery.query)}
-                        updateAggregation={(aggregation) => this.props.updateQueryAggregation(index, aggregation)}
-                        removeAggregation={canRemoveAggregation ? this.props.removeQueryAggregation.bind(null, index) : null}
+                        tableMetadata={query.tableMetadata()}
```

Comment:
So far I've only ported this file to the lib, so basically anywhere you see `query.tableMetadata()` and `query.datasetQuery()` is legacy code.

We can just pass `aggregation`, `query`, and `setDatasetQuery` into `AggregationWidget`.